src/osd/ECCommon.cc

```diff
@@ -589,8 +633,19 @@ int ECCommon::ReadPipeline::send_all_remaining_reads(
   if (r)
     return r;
 
-  list<boost::tuple<uint64_t, uint64_t, uint32_t> > offsets =
-    rop.to_read.find(hoid)->second.to_read;
+  list<ec_align_t> to_read = rop.to_read.find(hoid)->second.to_read;
+
+  if (cct->_conf->osd_ec_partial_reads) {
```

Comment:
Here also, I think it's probably simpler to consider ReadPipeline, as a whole, as accepting chunk aligned rather than stripe aligned offsets.  It's probably simpler to just rebuild new_to_read unconditionally.

Reply:
See my comment on refresh_complete.

Reply:
> and !fast_read?

Not necessarily. It's for preserving the old logic. For `fast_reads` the realignment should be already NOP.

Reply:
Although not strictly tied with the chunk alignment, dropping the chunk-to-stripe transformata would simplify the things.